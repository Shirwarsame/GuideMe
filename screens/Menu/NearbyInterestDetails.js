import React from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import { Icon, Button } from "native-base";


/**Prop passed
 * @param  {} navigation props.navigation is the name of the object from Navigator library
 */

function NearbyInterestDetails(props) {

    const goBack = () => {
        props.navigation.goBack();
    }

    const NearbyInterestDetailsScreen = true;
    const goToDoubleSearch = () => {
        props.navigation.navigate("DoubleSearch", { NearbyInterestDetailsScreen: NearbyInterestDetailsScreen });
    }

    const name = props.navigation.getParam("name", null);
    const rating = props.navigation.getParam("rating", null);
    const photoref = props.navigation.getParam("photoref", null);

    return (
        <View style={styles.container}>
           

            <View style={styles.imageContainer}>
                <Image style={styles.placeImage} source={{ uri: photoref }} />
            </View>
            <Text style={styles.mainLabel}>{name ? name : "N/A"}</Text>
            <Text style={styles.reviewLabel}>{rating ? rating : "N/A"}</Text>

            <View style={styles.backArrowContainer}>
                <Button transparent style={styles.backArrow} onPress={goBack}>
                    <Icon name="md-arrow-round-back" style={styles.icon}></Icon>
                </Button>
            </View>

            <Button transparent style={styles.routeButton} onPress={goToDoubleSearch}><Text style={styles.viewRouteText}>View Route</Text></Button>
        </View >
    );
}

export const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        justifyContent: "space-between",
        height: "100%",
        width: "100%",
        backgroundColor: "#2A2E43"
    },
    mainLabel: {
        color: "#FFF",
        fontSize: 23,
        fontFamily: "encodeSansExpanded",
        paddingBottom: 10,
        top: "14%",
        bottom: "2%",
        fontWeight: "bold",
        justifyContent: "center",
        position: "absolute"
    },
    icon: {
        position: "absolute",
        color: "#FFFFFF",
        alignSelf: "center",
        fontSize: 35
    },
    routeButton: {
        width: "90%",
        height: "8%",
        fontSize: 25,
        bottom: "8%",
        justifyContent: "center",
        backgroundColor: "#3ACCE1",
        borderRadius: 10,
    },
    viewRouteText: {
        color: "white",
        fontSize: 14,
    },
    backArrow: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
        left: "10%"
    },
    backArrowContainer: {
        width: "100%",
        height: "6%",
        flexDirection: "column",
        justifyContent: "space-around",
        alignContent: "center",
        alignItems: "center",
        top: "7%"
    },

    imageContainer: {
        width: "100%",
        height: "32%",
        top: "0%",
        position: "absolute",
        opacity: 0.3
    },

    placeImage: {
        width: "100%",
        height: "100%",
        top: "0%",
        position: "relative"
    },
    mainLabel: {
        color: "#FFFFFF",
        left: "5%",
        position: "absolute",
        fontSize: 30,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "19%"
    },
    subLabel: {
        position: "absolute",
        top: "24%",
        left: "5%",
        color: "#FFFFFF",
        fontSize: 11,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded"
    },
    reviewLabel: {
        position: "absolute",
        top: "27%",
        left: "5%",
        color: "#FFFFFF",
        fontSize: 20,
        fontFamily: "encodeSansExpanded"
    },

});

export default NearbyInterestDetails;
