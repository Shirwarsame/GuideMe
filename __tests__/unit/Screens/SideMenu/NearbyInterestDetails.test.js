import React from "react";
import NearbyInterestDetails from "../../../../screens/Menu/NearbyInterestDetails";
import renderer from "react-test-renderer";

describe("Nearby Interest Details component", () => {
    test("renders correctly", () => {
        const navigation = {
            getParam: (param, defaultValue) => {
              return "Tim Hortons"
            }
          };
        const tree = renderer.create(<NearbyInterestDetails navigation={navigation}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});