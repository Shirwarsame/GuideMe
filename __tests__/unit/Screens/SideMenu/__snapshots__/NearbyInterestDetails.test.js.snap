// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nearby Interest Details component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2A2E43",
      "height": "100%",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#000",
          "bottom": "78%",
          "height": "32%",
          "opacity": 0.3,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "100%",
            "position": "relative",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "bottom": "33%",
          "flexDirection": "column",
          "height": "6%",
          "justifyContent": "space-around",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "space-between",
            "left": "10%",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
            "width": "100%",
          }
        }
      >
        <Text />
      </View>
    </View>
    <Text
      style={
        Object {
          "bottom": "28%",
          "color": "#FFFFFF",
          "fontFamily": "encodeSansExpanded",
          "fontSize": 30,
          "fontWeight": "bold",
          "textAlign": "left",
          "width": "90%",
        }
      }
    >
      Tim Hortons
    </Text>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "bottom": "50%",
          "elevation": 0,
          "flexDirection": "row",
          "height": "8%",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "width": "16%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "bottom": "2%",
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 20,
              "left": "20%",
            }
          }
        >
          N/A
        </Text>
      </View>
      <View
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "width": "4%",
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "height": "100%",
            "justifyContent": "center",
            "width": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "height": "40%",
              "width": "60%",
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "bottom": "49%",
          "elevation": 0,
          "flexDirection": "row",
          "height": "8%",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#353A50",
            "borderRadius": 10,
            "height": "100%",
            "justifyContent": "center",
            "width": "16%",
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "width": "4%",
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "height": "100%",
            "justifyContent": "center",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 13,
              "position": "absolute",
            }
          }
        >
          N/A
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "bottom": "47%",
          "elevation": 0,
          "flexDirection": "row",
          "height": "8%",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#353A50",
            "borderRadius": 10,
            "height": "100%",
            "justifyContent": "center",
            "width": "16%",
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "width": "4%",
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "height": "100%",
            "justifyContent": "center",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 13,
              "position": "absolute",
            }
          }
        >
          Closed Now
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "bottom": "45%",
          "elevation": 0,
          "flexDirection": "row",
          "height": "8%",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#353A50",
            "borderRadius": 10,
            "height": "100%",
            "justifyContent": "center",
            "width": "16%",
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "width": "4%",
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "height": "100%",
            "justifyContent": "center",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 13,
              "position": "absolute",
            }
          }
        >
          N/A
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "bottom": "43%",
          "elevation": 0,
          "flexDirection": "row",
          "height": "8%",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#353A50",
            "borderRadius": 10,
            "height": "100%",
            "justifyContent": "center",
            "width": "16%",
          }
        }
      >
        <Text />
      </View>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "width": "4%",
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "height": "100%",
            "justifyContent": "center",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 13,
              "position": "absolute",
            }
          }
        >
          Coming Soon
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "bottom": "41%",
          "elevation": 0,
          "flexDirection": "row",
          "height": "8%",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#353A50",
            "borderRadius": 10,
            "height": "100%",
            "justifyContent": "center",
            "width": "16%",
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "width": "4%",
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "height": "100%",
            "justifyContent": "center",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 13,
              "position": "absolute",
            }
          }
        >
          Coming Soon
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3ACCE1",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "bottom": "8%",
          "elevation": 0,
          "flexDirection": "row",
          "fontSize": 25,
          "height": "8%",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 14,
          }
        }
      >
        Get Directions
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
